examples/text_summarization/pointer_summarizer/src/decode.py

```diff
@@ -0,0 +1,206 @@
+                             for t in latest_tokens]
+            y_t_1 = paddle.to_tensor(latest_tokens)
+            all_state_h =[]
+            all_state_c = []
+
+            all_context = []
+
+            for h in beams:
+                state_h, state_c = h.state
+                all_state_h.append(state_h)
+                all_state_c.append(state_c)
+
+                all_context.append(h.context)
+
+            s_t_1 = (paddle.stack(all_state_h, 0).unsqueeze(0), paddle.stack(all_state_c, 0).unsqueeze(0))
+            c_t_1 = paddle.stack(all_context, 0)
+
+            coverage_t_1 = None
+            if config.is_coverage:
+                all_coverage = []
+                for h in beams:
+                    all_coverage.append(h.coverage)
+                coverage_t_1 = paddle.stack(all_coverage, 0)
+
+            final_dist, s_t, c_t, attn_dist, p_gen, coverage_t = self.model.decoder(y_t_1, s_t_1,
+                                                        encoder_outputs, encoder_feature, enc_padding_mask, c_t_1,
+                                                        extra_zeros, enc_batch_extend_vocab, coverage_t_1, steps)
+            log_probs = paddle.log(final_dist)
+            topk_log_probs, topk_ids = paddle.topk(log_probs, config.beam_size * 2)
+
+            dec_h, dec_c = s_t
+            dec_h = dec_h.squeeze()
+            dec_c = dec_c.squeeze()
+
+            all_beams = []
+            num_orig_beams = 1 if steps == 0 else len(beams)
+            for i in range(num_orig_beams):
+                h = beams[i]
+                state_i = (dec_h[i], dec_c[i])
+                context_i = c_t[i]
+                coverage_i = (coverage_t[i] if config.is_coverage else None)
+
+                for j in range(config.beam_size * 2):  # for each of the top 2*beam_size hyps:
+                    new_beam = h.extend(token=topk_ids[i, j].numpy()[0],
+                                   log_prob=topk_log_probs[i, j].numpy()[0],
+                                   state=state_i,
+                                   context=context_i,
+                                   coverage=coverage_i)
+                    all_beams.append(new_beam)
+
+            beams = []
+            for h in self.sort_beams(all_beams):
+                if h.latest_token == self.vocab.word2id(data.STOP_DECODING):
+                    if steps >= config.min_dec_steps:
+                        results.append(h)
+                else:
+                    beams.append(h)
+                if len(beams) == config.beam_size or len(results) == config.beam_size:
+                    break
+
+            steps += 1
+
+        if len(results) == 0:
+            results = beams
+
+        beams_sorted = self.sort_beams(results)
+
+        return beams_sorted[0]
+
+if __name__ == '__main__':
+    model_filename = sys.argv[1]
+    beam_Search_processor = BeamSearch(model_filename)
```

Comment:
`beam_Search_processor` -> `beam_search_processor`
Please make sure use unified naming style for variables.